client/src/lib/components/NewExpenseDialog.svelte

```diff
@@ -0,0 +1,203 @@
+			$value.value = predefinedExpense.value + ''; // TODO solve discrepancy between predefined expense number value and expense string value
+			$currencyTypeId.value = predefinedExpense.currencyType.id;
+			$recurrenceTypeId.value = predefinedExpense.recurrenceType.id;
+		} else {
+			predefinedExpenseId = null;
+			$name.value = '';
+			$description.value = '';
+			$value.value = '';
+			$currencyTypeId.value = 0;
+			$recurrenceTypeId.value = 0;
+		}
+	}
+
+	export function show() {
+		open = true;
+	}
+</script>
+
+<Dialog bind:open aria-labelledby="title" aria-describedby="content" actions={null}>
+	<Title id="title">New Expense</Title>
+	<Content id="content">
+		<div class="form-container">
+			<Select
+				bind:value={predefinedExpenseId}
+				label="Predefined expense"
+				on:SMUIMenu:selected={() => onPredefinedExpenseSelect(predefinedExpenseId)}
+			>
+				<Option value={-1}>-</Option>
+				{#each predefinedExpenses as predefinedExpense}
+					<Option value={predefinedExpense.id}>{predefinedExpense.name}</Option>
+				{/each}
+			</Select>
+
+			<Textfield
+				variant="outlined"
+				bind:value={$name.value}
+				label="Name"
+				bind:dirty={$name.dirty}
+				bind:invalid={$name.invalid}
+				><svelte:fragment slot="helper">
+					{#if !$name.valid}
+						<HelperLine>Please enter the expense name</HelperLine>
+					{/if}
+				</svelte:fragment>
+			</Textfield>
+
+			<Textfield
+				textarea
+				variant="outlined"
+				bind:value={$description.value}
+				label="Description"
+				bind:dirty={$description.dirty}
+				bind:invalid={$description.invalid}
+				input$resizable={false}
+				><svelte:fragment slot="helper">
+					{#if !$name.valid}
+						<HelperLine>Please enter a description</HelperLine>
+					{/if}
+				</svelte:fragment>
+			</Textfield>
+
+			<Textfield
+				variant="outlined"
+				bind:value={$startDate.value}
+				label="Date"
+				bind:dirty={$startDate.dirty}
+				bind:invalid={$startDate.invalid}
+				><svelte:fragment slot="helper">
+					{#if !$startDate.valid}
+						<HelperLine>Please enter the expense start date</HelperLine>
+					{/if}
+				</svelte:fragment>
+			</Textfield>
+
+			<Textfield
+				variant="outlined"
+				bind:value={$value.value}
+				label="Value"
+				bind:dirty={$value.dirty}
+				bind:invalid={$value.invalid}
+				type="number"
+				><svelte:fragment slot="helper">
+					{#if !$value.valid}
+						<HelperLine>Please enter the expense value</HelperLine>
+					{/if}
+				</svelte:fragment>
+			</Textfield>
+
+			<Select bind:value={$currencyTypeId.value} label="Currency">
+				{#each currencyTypes as currency}
+					<Option value={currency.id}>{currency.abbreviation}</Option>
+				{/each}
+				<svelte:fragment slot="helperText">
+					{#if !$currencyTypeId.valid}
+						<HelperLine>Please select a currency</HelperLine>
+					{/if}
+				</svelte:fragment>
+			</Select>
+
+			<Select bind:value={$recurrenceTypeId.value} label="Currency">
```

Comment:
The label should be _Recurrence_ I guess

Reply:
Yep, fixed